src/auth/auth.ts

```diff
@@ -1,7 +1,16 @@
+/*
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *  SPDX-License-Identifier: Apache-2.0
+ */
+
 import { AxiosInstance } from 'axios';
 
 export default interface Auth {
     // Hook into axios and intercept requests to add Auth tokens
     // https://masteringjs.io/tutorials/axios/interceptors
-    attachInterceptor(axiosInstance: AxiosInstance): void;
+    attachInterceptor(axiosInstance: AxiosInstance, interceptorConfig: InterceptorConfig): void;
 }
+
+export type InterceptorConfig = {
+    awsRegion?: string;
+};
```

Comment:
I think that the `InterceptorConfig` type is not helping.

```ts
// ts valid code that fails(I think) at runtime.
IamAuth().attachInterceptor(axiosInstance, {})
```
Is `awsRegion` truly optional? What happens if you pass undefined as region to `aws4Interceptor`? 
Also, this means that **every** Auth implementation accepts awsRegion as a config option. Is that what we want?

Reply:
I was trying to find a way to allow the `Auth` interface to handle different use cases. For example, how to handle an `Auth` implementation that only requires a `region`. How to handle an `Auth` implementation that only requires an `api-key`. 

After further thought, I think the best way to do that is not through the interface, but through the `constructor` of the class implementing `Auth`. In this case `IamAuth` requires a region, so we'll pass the region to the class through the constructor. Other classes that implement `Auth` can pass in whichever params they require through their own constructor.

Reply:
Much better